engine/crates/integration-tests/tests/federation/auth/multiple.rs

```diff
@@ -0,0 +1,73 @@
+use engine_v2::Engine;
+use integration_tests::federation::GraphqlResponse;
+use integration_tests::openid::OrkyHydraOpenIDProvider;
+use integration_tests::{
+    federation::EngineV2Ext,
+    mocks::graphql::FakeGithubSchema,
+    openid::{AUDIENCE, ISSUER, JWKS_URI},
+    runtime, MockGraphQlServer,
+};
+use openidconnect::OAuth2TokenResponse;
+
+#[test]
+fn test_provider() {
+    runtime().block_on(async move {
+        let github_mock = MockGraphQlServer::new(FakeGithubSchema).await;
+
+        let engine = Engine::builder()
+            .with_schema("github", &github_mock)
+            .await
+            .with_supergraph_config(format!(
+                r#"extend schema @authv2(providers: [
+                    {{ name: "my-oauth2", type: "oauth2", issuer: "{ISSUER}", jwt: {{ audience: "{AUDIENCE}" }} }}
+                    {{ name: "my-jwt", type: "jwt", jwks: {{ url: "{JWKS_URI}" }} }}
+                ])"#
+            ))
+            .finish()
+            .await;
+        let provider = OrkyHydraOpenIDProvider::default();
+        let client = provider.create_client().await;
+
+        // OAuth2 will refuse the token without audience, but JWT will happily accept it, because
+        // who cares?
```

Comment:
This test relies on the implementation detail of JWT ignoring audiences.  So if we ever change that (and I think we should - ignoring audiences seems bad) this test is made less valid.

Might be better to have two actually different providers for this?

Reply:
yeah was a lazy attempt to not add even more stuff in the docker compose